src/MainUI.cpp

```diff
@@ -157,57 +143,59 @@ void MainUI::openModuleUI(Wt::WMenuItem* gitem) {
-    // }
-    redi::ipstream proc("export GRASS_PNG_AUTO_WRITE=TRUE; export GRASS_PNG_COMPRESSION=9; export GRASS_TRANSPARENT=TRUE; export GRASS_TRUECOLOR=TRUE; export LD_LIBRARY_PATH=/usr/lib/grass70/lib; export GISBASE=/usr/lib/grass70/; export GISDBASE=/home/mayank/grassdata1; export GISRC=/home/mayank/.grass7/rc; export PATH=/usr/lib/grass70/bin:/usr/lib/grass70/scripts:$PATH; g.gisenv");
-    // redi::ipstream proc("export GRASS_PNG_AUTO_WRITE=TRUE; export GRASS_PNG_COMPRESSION=9; export GRASS_TRANSPARENT=TRUE; export GRASS_TRUECOLOR=TRUE; export LD_LIBRARY_PATH=/usr/lib/grass70/lib; export GISBASE=/usr/lib/grass70/; export GISDBASE=/home/mayank/grassdata1; export GISRC=/home/mayank/.grass7/rc; export PATH=/usr/lib/grass70/bin:/usr/lib/grass70/scripts:$PATH; g.guienv");
-    // redi::ipstream proc("g.guienv");      
-      std::string line;
-      std::cout.flush();
-//       while (proc >> line) {
-//     std::cout << line << std::endl;
-// }
-      while (std::getline(proc.out(), line)){
-        std::cout << "stdout: " << line << '\n';
-      }
-
-      // redi::ipstream proc1("ls");  
-      // std::string line1;
-      // std::cout.flush();
-      // while (std::getline(proc1.out(), line1)){
-      //   std::cout << "stdout: " << line1 << '\n';
-      // }
+  if (code == Wt::WDialog::Accepted)
+  {
+        std::cout<<(mod->findById("Main"))->objectName()<<"check"<<endl;
+        std::map<std::string,std::vector<Parameter *> > mop = mod->map;
+        std::vector<std::string>list= mod->container_IDs;
+
+        std::string command = gmodule+" ";
+        for (std::vector<std::string>::iterator it=list.begin(); it != list.end(); ++it) /*creation of widgets*/
+                 {   
+                    
+
+                      std::string d = *it;
+                      // std::cout<<d<<endl;
+                      WLineEdit *edit = dynamic_cast<WLineEdit *>(((mod->findById("Main")))->findById(d));
+                      // std::cout<<(edit->text()).toUTF8()<<std::endl;
+                      std::string value = (edit->text()).toUTF8();
+                      command = command + d+"="+value + " ";
+
+                    
+
+                 }
+        std::cout<<command<<std::endl;
+            // redi::ipstream proc("grass70 $HOME/grassdata/newLocation/PERMANENT", redi::pstreams::pstdout | redi::pstreams::pstdin );
+            //   std::string line;
+            //   std::cout.flush();
+            //   while (std::getline(proc.out(), line)){
+            //     std::cout << "stdout: " << line << '\n';
+            // }
+            redi::ipstream proc("export GRASS_PNG_AUTO_WRITE=TRUE; export GRASS_PNG_COMPRESSION=9; export GRASS_TRANSPARENT=TRUE; export GRASS_TRUECOLOR=TRUE; export LD_LIBRARY_PATH=/usr/lib/grass70/lib; export GISBASE=/usr/lib/grass70/; export GISDBASE=/home/mayank/grassdata1; export GISRC=/home/mayank/.grass7/rc; export PATH=/usr/lib/grass70/bin:/usr/lib/grass70/scripts:$PATH; g.gisenv");
```

Comment:
what if you do those export in shell script and then running grass commands such as g.list rast ?